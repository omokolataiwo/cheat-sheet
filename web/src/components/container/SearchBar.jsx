import React from 'react';
import TextInputField from './TextInputField';

const SearchBar = ({ onSearchSheetCheat }) => (
  <div>
    <TextInputField
      name="firstName"
      placeholder="Search Sheet Cheat"
      type="text"
      size="3"
      onChange={onSearchSheetCheat}
    />
  </div>
);

export default SearchBar;
