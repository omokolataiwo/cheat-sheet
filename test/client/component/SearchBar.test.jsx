import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../../../web/src/components/container/SearchBar';

const props = {};
const wrapper = shallow(<SearchBar {...props} />);

describe('Search Bar Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
