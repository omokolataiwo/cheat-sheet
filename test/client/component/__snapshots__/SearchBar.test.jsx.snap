// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <TextInputField
        errors={Array []}
        icon="search"
        label="Search Git Cheat Sheet"
        name="search-input"
        placeholder="Search Sheet Cheat"
        type="text"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "errors": Array [],
        "icon": "search",
        "label": "Search Git Cheat Sheet",
        "name": "search-input",
        "onChange": undefined,
        "placeholder": "Search Sheet Cheat",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TextInputField
          errors={Array []}
          icon="search"
          label="Search Git Cheat Sheet"
          name="search-input"
          placeholder="Search Sheet Cheat"
          type="text"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "errors": Array [],
          "icon": "search",
          "label": "Search Git Cheat Sheet",
          "name": "search-input",
          "onChange": undefined,
          "placeholder": "Search Sheet Cheat",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
